### ISSUE TITLE
Lens + Helm broken on macOS with overridden `XDG_*` env vars

### ISSUE THREAD

**Issue by chasewalden** [kontena/lens#4933]
**Describe the bug**
If using overridden `XDG_*`, the Apps/Charts view and Preferences/Kubernetes/Helm Charts views seem to break.
This appears to be due to how the helm binary is executed.

When adding Helm Repos via the Preferences GUI, the repo is not added when running `helm repo list` in either the built in terminal, or Terminal.app. Adding a repo via CLI does not reflect in the Preferences GUI.

In either case, adding a repo via GUI or CLI causes the Apps/Charts view to only list the default Bitnami Charts and no other repos can be seen.

Unsetting/commenting out the `XDG_*` lines and re-launching Lens fixes the behavior. CLI and GUI both work and Apps/Charts now shows added repos.

**To Reproduce**
1. On macOS with `zsh` as default shell (standard since macOS Catalina), override `XDG_*` vars in `.zshrc`, or `.zshenv`
2. Re-launch Lens
3. Open Preferences/Kubernetes and add a new Helm Repo
4. Connect to any Cluster and open Apps/Charts

**Expected behavior**
Charts from the new repo are listed and can be installed via GUI.

**Environment (please complete the following information):**
- Lens Version: 5.3.4-latest.20220120.1
- OS: macOS Big Sur 11.6.4
- Installation method (e.g. snap or AppImage in Linux): .dmg from k8slens.dev

**Additional context**

On my machine, `zsh` is my default shell and I have added the following to my `.zshenv`
```sh
# file: ~/.zshenv

# align with typical *nix systems
export XDG_CONFIG_HOME="$HOME/.config"
export XDG_CACHE_HOME="$HOME/.cache"
export XDG_DATA_HOME="$HOME/.local/share"
export XDG_STATE_HOME="$HOME/.local/state"
```

**Comment by chasewalden** [comment]
It is strange that this causes a bug since it seems like both the Apps/Charts view and Preferences/Kubernetes/Helm Charts views use the HelmCLI class to execute helm, which means that they should both be using the same locations for reading/writing data in the GUI, even though it is not synced with the shell (integrated or external).

Looking at the code, it seems like this should work (at least through GUI) regardless of the XDG vars, but there must be a mismatch somewhere between the repo-manager and the chart-manager.

nodejs child_process runs `/bin/sh` as the default shell, but macOS confusingly doesn't symlink to zsh despite it being the default. 

This is probably a bug that affects a small number of users. I think most people don't know how/want to customize the XDG values. It might be useful to add an `Advanced` preference pane for users to customize certain values. 

One idea is to execute `helm env` on first use to get the `HELM_*_HOME` vars. Alternatively, these could be defaulted to some internal path that is local to Lens itself. These values can be saved in a user preferences file and injected into the child_process.exec call and built-in terminal, similar to the KUBECONFIG var. These values could be overridden in the Advanced pane. The helm cli docs describe the cache/config/data path resolution here: https://helm.sh/docs/helm/helm/

**Comment by Nokel81** [comment]
Note we do actually execute `helm env` and parse its values.

**Comment by chasewalden** [comment]
@Nokel81 yes, but the values emitted from `helm env` change depending on some other environment variables. In this case, my overloaded XDG variables in my usual shell change helm’s directories and appear to break the integration when using child_process. 

To maybe rephrase my previous suggestion: rather than parse `helm env` every time, parse it only once when Lens is first executed to populate the user preferences file and use the saved values when setting the child process env
